src/nvim/event/proc.c

```diff
@@ -437,6 +437,17 @@ static void on_proc_exit(Proc *proc)
   Loop *loop = proc->loop;
   ILOG("child exited: pid=%d status=%d" PRIu64, proc->pid, proc->status);
 
+#ifdef MSWIN
+  // XXX: This assumes the TUI never spawns any other processes...?
+  // TODO(justinmk): figure out why rpc_close sometimes(??) isn't called, then remove this jank.
+  // Theories:
+  // - EOF not received in receive_msgpack, then doesn't call chan_close_on_err().
+  // - proc_close_handles not tickled by ui_client.c's LOOP_PROCESS_EVENTS?
```

Comment:
For reference: https://github.com/neovim/neovim/issues/33019#issuecomment-2749523625 confirmed that the `nvim --embed` server terminates, only the UI process hangs.